app/controllers/patients_controller.rb

```diff
@@ -451,8 +451,10 @@ def bulk_update
     patients = current_user.get_patients(patient_ids)
 
     patients.each do |patient|
-      update_monitoring_fields(patient, params, non_dependent_patient_ids.include?(patient[:id]) ? :patient : :dependent,
-                               params[:apply_to_household] ? :group : :none)
+      # We never want to update closed records monitoring status via the bulk_update
+      update_params = patient.monitoring ? params : params.except('monitoring')
```

Comment:
Do these params need to be manipulated on each patient in the bulk update?

Reply:
Yes because we only want this to be applied to those monitorees who are not monitoring.

Reply:
`Benchmark.bmbm` with 10k iterations
```              user     system      total        real
remove    0.006958   0.000071   0.007029 (  0.007050)
nothing   0.000760   0.000025   0.000785 (  0.000775)
```

IPS report
```
Warming up --------------------------------------
              remove   200.730k i/100ms
             nothing     1.443M i/100ms
Calculating -------------------------------------
              remove      1.853M (± 7.0%) i/s -      9.234M in   5.009284s
             nothing     13.157M (± 3.6%) i/s -     66.357M in   5.050083s

Comparison:
             nothing: 13156928.8 i/s
              remove:  1852505.9 i/s - 7.10x  (± 0.00) slower
```

Code:
```ruby
module Bench

  @test_data = {hash: 'hash', of: 'of', params: 'params'}

  def nothing
    @test_data
  end

  def remove
    @test_data.except(:of)
  end

  def benchmark
    n = 10_000
    Benchmark.bmbm do |x|
      x.report('remove') { n.times do; remove; end }
      x.report('nothing') { n.times do; nothing; end }
    end
  end

  def ips
    Benchmark.ips do |x|
      x.report('remove') { remove }
      x.report('nothing') { nothing }
      x.compare!
    end
  end
end
```